### PR TITLE
Add Document.title and TitleChangedEvent

### PR DESCRIPTION
**Pull request by havocp**

### CODE REVIEW

**Comment by bryevdv** [comment]
A few minor comments but otherwise LGTM (have not checked out and tested yet)

**Comment by havocp** [comment]
If you think we should s/force_roundtrip/_force_roundtrip/g I can do that in its own PR covering both the coffee and python clients, perhaps.

**Comment by havocp** [comment]
ready to look over again, @bryevdv

File: bokeh/server/protocol/messages/patch_doc.py [hunk 1]
```diff
@@ -66,6 +66,11 @@ def should_suppress_on_change(self, event):
                 if event_json['kind'] == 'RootRemoved' and \
                    event_json['model']['id'] == event.model._id:
                     return True
+        elif isinstance(event, TitleChangedEvent):
+            for event_json in self.content['events']:
+                if event_json['kind'] == 'TitleChanged' and \
+                   event_json['title'] == event.title:
+                    return True
```
Comment (fpliger) [inline bokeh/server/protocol/messages/patch_doc.py]: What about moving the `_should_suppress_on_change` logic of each specific event to their classes? Feel like taxonomy of events and conditions can quickly make this method blow (in readability and maintainability)..

Comment (havocp) [inline bokeh/server/protocol/messages/patch_doc.py]: Good idea indeed. Can I add it to the todo list and make that its own PR?

Comment (fpliger) [inline bokeh/server/protocol/messages/patch_doc.py]: Sounds good. I have the callbacks PR coming through too so it's actually even better :)

**Comment by fpliger** [comment]
LGTM
